scipy/special/_logsumexp.py

```diff
@@ -203,9 +203,24 @@ def _logsumexp(a, b, axis, return_sign, xp):
     m = (xp.sum(i_max_dt, axis=axis, keepdims=True, dtype=a.dtype) if b is None
          else xp.sum(b * i_max_dt, axis=axis, keepdims=True, dtype=a.dtype))
 
-    # Arithmetic between infinities will introduce NaNs.
-    # `+ a_max` at the end naturally corrects for removing them here.
-    shift = xp.where(xp.isfinite(a_max), a_max, 0.)
+
+    # `a - shift` will introduce NaNs if both `a` and `shift` have infinities of the
+    # same sign.
+    # For positive infinite `shift`, `a - shift` does the right thing if all of `a` are
+    # finite. If some of `a` are imaginary with `+inf` real component, it is not a huge
+    # problem for NaNs to appear. The *best* `logsumexp` result in this case could have
+    # either `inf` or `nan` real part and finite or `nan` imaginary part, depending on
+    # the specifics. The logic for this is not implemented, and previous implementations
+    # of `logsumexp` have returned garbage phase for multiple complex infinities, so
+    # returning `nan + nan*j` is an improvement over a result with incorrect phase.
+    # For negative infinities, this can only occur if *all* elements have negative
+    # infinite real part, so the result is the logarithm of `0`, and the phase is
+    # meaningless. The value returned by `logsumexp` will have the same phase as
+    # `a_max`, which should be fine, since this function has never had a principled
+    # phase convention for this case. Perhaps NaN phase would be best, but we save
+    # that for another day.
+    shift = xp.where(xp.logical_or(xp.isfinite(a_max), xp_real(a_max) > 0),
```

Comment:
```suggestion
    shift = xp.where(xp.logical_or(xp.isfinite(a_max), xp.real(a_max) > 0),
```

I missed this one when resolving the merge conflicts.